src/ecs/util/systems.rs

```diff
@@ -0,0 +1,22 @@
+use util::circular_buffer::CircularBuffer;
+use util::time;
+///FPSCounterSystem
+///Add this system to your game to calculate FPS
+pub struct FPSCounterSystem;
+impl<'a> System<'a> for FPSCounterSystem {
+    type SystemData = (
+     Fetch<'a, Time>,
+     FetchMut<'a, FPSCounter>,
+     );
+    fn run(
+        &mut self,
+        (
+         time,
+         mut counter,
+         ):
+         Self::SystemData,
+    ){
+        counter.push(duration_to_secs(time.delta_time));
```

Comment:
Directly storing the `Duration` would be nicer imo.

Reply:
As parameter of counter.push or as a variable in the system?

Reply:
@msiglreith

Reply:
Sry for the delayed response. Storing `Duration`s inside the `FpsCounter` instead of `f32`.

Reply:
Lost of memory, lost of performance.
I'd have to put the durations back into f32 each time I want the average fps each tick.

Reply:
Also, I doubt anyone will ever interact directly with it without using the System

Reply:
You would be calculating the average of a u64 and a u32 number, and converting to f32 once in `sampled_fps`. I don't see how that is less performant, especially when the user doesn't request the sampled fps every frame.

Reply:
Actually you can store single u64 that represents nanoseconds. u64 can store more than hundred years of nanoseconds.
30 years is about 1'000'000'000'000'000'000 nanoseconds